generate-plugin.js

```diff
@@ -0,0 +1,134 @@
+'use strict'
+
+const {
+  readFile,
+  writeFile
+} = require('fs').promises
+const { existsSync } = require('fs')
+const path = require('path')
+const chalk = require('chalk')
+const generify = require('generify')
+const argv = require('yargs-parser')
+const cliPkg = require('./package')
+const { execSync } = require('child_process')
+const { promisify } = require('util')
+const log = require('./log')
+
+const pluginTemplate = {
+  dir: 'plugin',
+  main: 'index.js',
+  types: 'index.d.ts',
+  scripts: {
+    lint: 'standard && npm run lint:typescript',
+    'lint:typescript': 'ts-standard',
+    test: 'npm run lint && npm run unit && npm run test:typescript',
+    'test:typescript': 'tsd',
+    unit: 'tap test/**/*.test.js'
+  },
+  dependencies: {
+    'fastify-plugin': cliPkg.devDependencies['fastify-plugin']
+  },
+  devDependencies: {
+    '@types/node': cliPkg.devDependencies['@types/node'],
+    fastify: cliPkg.devDependencies.fastify,
+    standard: cliPkg.devDependencies.standard,
+    tap: cliPkg.devDependencies.tap,
+    'ts-standard': cliPkg.devDependencies['ts-standard'],
+    tsd: cliPkg.devDependencies.tsd,
+    typescript: cliPkg.devDependencies.typescript
+  },
+  tsd: {
+    directory: 'test'
+  },
+  logInstructions: function (pkg) {
+    log('debug', 'saved package.json')
+    log('info', `project ${pkg.name} generated successfully`)
+    log('debug', `run '${chalk.bold('npm install')}' to install the dependencies`)
+    log('debug', `run '${chalk.bold('npm test')}' to execute the tests`)
+  }
+}
+
+async function generate (dir, template) {
+  const generifyPromise = promisify(generify)
+  try {
+    const file = await generifyPromise(
+      path.join(__dirname, 'templates', template.dir),
+      dir,
+      {}
+    )
+    log('debug', `generated ${file}`)
+  } catch (err) {
+    throw new Error(err)
+  }
+  process.chdir(dir)
+  execSync('npm init -y')
+
+  log('info', `reading package.json in ${dir}`)
+  let data
+  try {
+    data = await readFile('package.json')
```

Comment:
On this case, I would use something like:

```suggestion
    pkg = await readFile('package.json').then(JSON.parse)
```
or
```suggestion
    packageString = await readFile('package.json')
    pkg = JSON.parse(packageString)
```